Feat: implement createGroup api client

kanta1207 opened the pull request:
<!--

Title Naming Conventions:

- Start with a capitalized prefix followed by an imperative form of a verb, just like a commit message.
- Refer to the "Commit Convention" for the prefix:
  https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#commit-convention.

Example:
- Fix: Resolve memory leak in data processing module
- Feat: Add two-factor authentication
- Refactor: Improve state management in React components
- Docs: Add Windows setup instructions

-->

## Overview

<!-- Provide a brief description of the changes introduced by this PR. -->
- Related issue : #181
- Implement `createGroup` api client, which works in client component
- Invoke `createGroup` method in `src/features/groups/components/CreateGroupButton/CreateGroupDrawerContent.tsx` with pure html `form` tag for now.
## Changes

- Add `createGroup` method to `src/lib/api/group/client/groupApiClient.client.ts`
- Add html `<form>` tag to `src/features/groups/components/CreateGroupButton/CreateGroupDrawerContent.tsx`
- Invoke `createGroup` in `src/features/groups/components/CreateGroupButton/CreateGroupDrawerContent.tsx`

## What I didn't in this PR
**I will tackle on Implementation of `CreateGroupForm` component in next scope, using common `Form` component and react-hook-form.** 
In this PR, I only focused on the API call.

## Review points

<!-- List the points you would like to be reviewed in detail and the points you are not confident about. -->
<!-- Delete this section if not needed -->

- Is API client `createGroup` method properly implemented with its test

## Assignee Checklist:

<!-- You can tick the checkboxes if not applicable -->

- [x] The base branch is correct (See: [Types of Branches](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

<!-- You can tick the checkboxes if not applicable -->

- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/genesis-tech-tribe/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] Code is well-commented and easy to understand
- [ ] UI changes accurately reflect the provided design

kanta1207 commented:
Importing `createGroupAction' into client component causing a build error.

kotaaaa reviewed (approved):
I thought this PR's scope is a little bit small, but it's good to go. Thanks!

uskayyyyy commented on src/features/groups/components/CreateGroupButton/CreateGroupDrawerContent.tsx:
```diff
@@ -8,34 +8,45 @@ import {
-  handleCreateGroup: () => void;
-}
+export const CreateGroupDrawerContent = () => {
+  const [groupName, setGroupName] = useState('');
```
Could you add `<IGroup['name']>`, please?
Currently, the type is inferred as a simple `string`, not referring to the type of group name.

uskayyyyy commented on src/lib/api/group/client/groupApiClient.client.ts:
```diff
@@ -1,39 +1,38 @@
+ * @property {string} groupId - The unique identifier of the group.
+ */
+export interface ICreateGroupApiResponse {
+  groupId: string;
```
`IGroup['id']`

uskayyyyy commented on src/lib/api/group/client/groupApiClient.client.ts:
```diff
@@ -1,39 +1,38 @@
+ * @property {string} groupName - The name of the group.
+ */
+export interface ICreateGroupParams {
+  groupName: string;
```
`IGroup['name']`

uskayyyyy reviewed (changes requested):
@kanta1207 
Thank you for the implementation! Please kindly check the comments I left.

kanta1207 commented:
> @kanta1207 Thank you for the implementation! Please kindly check the comments I left.

Thank you for your review! I addressed the points you brought up. Please take a relook at them 🙏

uskayyyyy reviewed (approved):
@kanta1207 
Thank you for updating! LGTM :)

kanta1207 commented:
> I thought this PR's scope is a little bit small, but it's good to go. Thanks!

@kotaaaa Yeah indeed maybe I tried to make it too small 😅 
I'll just replace `form` with `Form` component quickly in next PR
